sql/core/src/test/scala/org/apache/spark/sql/execution/joins/BroadcastJoinSuite.scala

```diff
@@ -0,0 +1,94 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TODO: uncomment the test here! It is currently failing due to
```

Comment:
Context for other reviewers: this test fails because it creates a new SparkContext which conflicts with the global TestSQLContext.  The test in question needs to run with `local-cluster` mode (to avoid some optimizations made for broadcast variables in the driver), which is why we can't re-use the global TestSQLContext. As a result, this test is commented out until we figure out how to eliminate the global TestSQLContext or otherwise provide hooks for managing its lifecycle.

@andrewor14, if we had a way to stop and restart the global TestSQLContext then we might be able to use that here: before we create the new context, stop the global one, and restore the global one at the end of the test.  This is a little bit hacky but may be better than what we have now.  Only subtlety here may relate to other global classes, such as TestData.

/cc @rxin, who maybe has some suggestions for quick-fixes here.